clients/cli/src/wrap.rs

```diff
@@ -0,0 +1,196 @@
+use {
+    crate::{
+        common::{parse_pubkey, parse_token_program, process_transaction},
+        config::Config,
+        output::{format_output, println_display},
+        CommandResult,
+    },
+    clap::Args,
+    serde_derive::{Deserialize, Serialize},
+    serde_with::{serde_as, DisplayFromStr},
+    solana_cli_output::{display::writeln_name_value, QuietDisplay, VerboseDisplay},
+    solana_pubkey::Pubkey,
+    solana_remote_wallet::remote_wallet::RemoteWalletManager,
+    solana_signature::Signature,
+    solana_signer::Signer,
+    solana_transaction::Transaction,
+    spl_token_wrap::{get_wrapped_mint_address, get_wrapped_mint_authority, instruction::wrap},
+    std::{
+        fmt::{Display, Formatter},
+        rc::Rc,
+        sync::Arc,
+    },
+};
+
+#[derive(Clone, Debug, Args)]
+pub struct WrapArgs {
+    /// The address of the token program that the unwrapped mint belongs to
+    #[clap(value_parser = parse_token_program)]
+    pub unwrapped_token_program: Pubkey,
+
+    /// The address of the mint to wrap
+    #[clap(value_parser = parse_pubkey)]
+    pub unwrapped_mint: Pubkey,
+
+    /// The address of the unwrapped token account to wrap from
+    #[clap(value_parser = parse_pubkey)]
+    pub unwrapped_token_account: Pubkey,
+
+    /// The address of the escrow account that will hold the unwrapped tokens
+    #[clap(value_parser = parse_pubkey)]
+    pub escrow_account: Pubkey,
+
+    /// The address of the token program that the wrapped mint should belong to
+    #[clap(value_parser = parse_token_program)]
+    pub wrapped_token_program: Pubkey,
+
+    /// The address of the token account to receive wrapped tokens
+    #[clap(value_parser = parse_pubkey)]
+    pub recipient_token_account: Pubkey,
+
+    /// The amount of tokens to wrap
+    #[clap(value_parser = |s: &str| -> Result<u64, String> { s.parse::<u64>().map_err(|e| e.to_string()) })]
```

Comment:
You shouldn't need to be this explicit, right?

```suggestion
    #[clap(value_parser)]
```